src/main.cpp

```diff
@@ -2826,6 +2848,102 @@ bool AbortNode(const std::string &strMessage, const std::string &userMessage) {
     return false;
 }
 
+
+/**
+ * BLOCK PRUNING CODE ("Autoprune")
+ */
+
+/* Calculate the amount of disk space the block & undo files currently use */
+uint64_t CalculateCurrentUsage()
+{
+    uint64_t retval=0;
+    BOOST_FOREACH(const CBlockFileInfo &file, vinfoBlockFile) {
+        retval += file.nSize + file.nUndoSize;
+    }
+    return retval;
+}
+
+/* Prune (delete) a block file.  Returns true if a file was deleted, else false */
+void PruneOneBlockFile(const int fileNumber)
+{
+    if (!fHavePruned) {
+        AssertLockHeld(cs_main);
+        pblocktree->WriteFlag("prunedblockfiles", true);
+        fHavePruned = true;
+    }
+    for (BlockMap::iterator it = mapBlockIndex.begin(); it != mapBlockIndex.end(); ++it) {
+        CBlockIndex* pindex = it->second;
+        if (pindex->nFile == fileNumber){
+            pindex->nStatus &= ~BLOCK_HAVE_DATA;
+            pindex->nStatus &= ~BLOCK_HAVE_UNDO;
+            pindex->nFile = 0;
+            pindex->nDataPos = 0;
+            pindex->nUndoPos = 0;
+            setDirtyBlockIndex.insert(pindex);
+        }
+    }
+    CDiskBlockPos pos(fileNumber, 0);
+    try 
+    {
+        boost::filesystem::remove(GetBlockPosFilename(pos, "blk"));
+        boost::filesystem::remove(GetBlockPosFilename(pos, "rev"));
+    }
+    catch (const std::runtime_error& e)
+    {
+        AbortNode("Fatal error removing block/undo file. Try running with -reindex");
+    }
+
+    vinfoBlockFile[fileNumber].SetNull();
+    setDirtyFileInfo.insert(fileNumber);
+}
+
+/**
+ * Autoprune()
+ *
+ * nAdditionalBytes, if supplied, is the number of bytes that
+ * are about to be allocated to write new blocks to disk.
+ */
+void Autoprune(uint64_t nAdditionalBytes)
+{
+    LOCK(cs_main);
+    if (chainActive.Tip() == NULL || nAutopruneTarget == 0) {
+        LogPrintf("Autoprune: No chain, or called when target=0.\n");
+        return;
+    }
+    if (chainActive.Tip()->nHeight <= Params().AutopruneAfterHeight()) {
+        LogPrintf("Autoprune: current height less than minimum height for pruning (%d).\n", Params().AutopruneAfterHeight());
```

Comment:
This seems like a pretty normal condition during initial sync; does that require an unconditional log message?